src/Message.php

```diff
@@ -59,15 +53,24 @@ public function aggregateRootId(): ?AggregateRootId
         return $this->headers[Header::AGGREGATE_ROOT_ID] ?? null;
     }
 
-    public function timeOfRecording(): PointInTime
+    public function timeOfRecording(): DateTimeImmutable
     {
-        return PointInTime::fromString($this->headers[Header::TIME_OF_RECORDING]);
+        /* @var DateTimeImmutable */
+        $timeOfRecording = DateTimeImmutable::createFromFormat(
+            self::TIME_OF_RECORDING_FORMAT,
+            $this->headers[Header::TIME_OF_RECORDING] ?? ''
+        );
+
+        assert(
```

Comment:
I was conflicted about this one. Generally, this should be configuration, as in the default headers decorator should be used. But then again, if it wasn't and people would add this, they could get into a situation where this would indeed pop up in production. I'll introduce an exception for it.